docs/core/v3.0.rst

```diff
@@ -0,0 +1,1761 @@
+extension               metadata                                  is extension required
+======================= ========================================= =====================
+generic                 ``extensions`` in `entry point metadata`_ ``must_understand``
+array                   ``extensions`` in `Array metadata`_       ``must_understand``
+data type               `data_type`_                              no ``fallback``
+chunk grid              `chunk_grid`_                             always
+chunk memory layout     `chunk_memory_layout`_                    always
+storage transformer     `storage_transformers`_                   always
+======================= ========================================= =====================
+
+There are no group extensions in Zarr v3.0.
+
+See https://github.com/zarr-developers/zarr-specs/issues/49 for a list of potential extensions
+
+Implementation Notes
+====================
+
+This section is non-normative and presents notes from implementers about cases
+that need to be carefully considered but do not strictly fall into the spec.
+
+Explicit vs implicit group
+--------------------------
+
+While the zarr spec v3 defined implicit and explicit group, implementation may
+decide to create an explicit group for all implicit group they encounter; in
+particular when using a hierarchical storage.
+
+Erasure of an implicit group may automatically erase any empty parent. For
+example on a S3 store where the namespace is flat, erasure of the last key with
+a prefix will erase all the implicit group in the prefix.
+
+Care must thus be taken when erasing a array or a group if the parent needs to
+be converted into an explicit group.
+
+
+Comparison with Zarr v2
+=======================
+
+This section is informative.
+
+Below is a summary of the key differences between this specification
+(v3) and Zarr v2.
+
+- In v3 each hierarchy has an explicit root, and must be opened at the
+  root. In v2 there was no explicit root and a hierarchy could be
+  opened at its original root or at any sub-group.
+
+- In v3 the storage keys have been redesigned to separate the space of
+  keys used for metadata and data, by using different prefixes. This
+  is intended to allow for more performant listing and querying of
+  metadata documents on high latency stores. There are also
+  differences including a change to the default separator used to
+  construct chunk keys, and the addition of a key suffix for metadata
+  keys.
+
+- v3 has explicit support for extensions via defined
+  extension points and mechanisms.
+
+- v3 allows for greater flexibility in how groups and arrays are
+  created. In particular, v3 supports implicit groups, which are
+  groups that do not have a metadata document but whose existence is
+  implied by descendant nodes. This change enables multiple arrays to
+  be created in parallel without generating any race conditions for
```

Comment:
This implies that implicit groups remove race conditions for parallel array creation, but isn't there still a potential race condition when trying to create the folder corresponding to the group, at least when having a nested layout on a filesystem?
E.g. if arrays `g/a` and `g/b` are created in an implicit group `g` that doesn't exist prior?
(Not sure if this is the right place to raise this, but I didn't find a sub-heading for implicit groups)

Reply:
> I guess so, but it should be much easier to fix than parallel writes to the metadata.

Yes, that makes sense. I just stumbled over this because "any race condition" sounded too strong; the new wording sounds good.